engine/crates/federated-graph/src/from_sdl.rs

```diff
@@ -138,9 +138,7 @@ fn ingest_fields<'a>(parsed: &'a ast::ServiceDocument, state: &mut State<'a>) ->
     for definition in &parsed.definitions {
         match definition {
             ast::TypeSystemDefinition::Schema(_) => {
-                return Err(DomainError(
-                    "Not implemented: schema definitions in federated schema".to_owned(),
-                ))
+                continue;
```

Comment:
What's the thinking here?  Seems to me like this error would be useful rather than silently ignoring schema definitions and potentially doing the wrong thing

Reply:
Fair point. My goal was to ignore the following:
https://github.com/the-guild-org/gateways-benchmark/blob/main/federation-v1/gateways/apollo-router/supergraph.graphql#L1

```
schema
  @link(url: "https://specs.apollo.dev/link/v1.0")
  @link(url: "https://specs.apollo.dev/join/v0.3", for: EXECUTION)
{
  query: Query
}
```

I didn't want to bother it more than necessary. I could just ignore `@link` directives and handle the custom query names though. Ok will do that

Reply:
FWIW I think that the composition implementations I've looked at (including ours) will always name the composed query type `Query`, the composed mutation type `Mutation` and so on and so on.

Reply:
Agreed it's most likely useless today, probably forever. Anyway did it quickly, ignoring `@link` directive and raising an error for other directives.